index.js

```diff
@@ -186,31 +190,29 @@ class Squeakquel extends Datastore {
     _defineTable(modelName) {
         const schema = MODELS[modelName];
         const tableName = `${this.prefix}${schema.tableName}`;
-        const fields = schema.base.describe().children;
+        const fields = schema.base.describe().keys;
         const tableFields = {};
         const tableOptions = {
             timestamps: false,
             indexes: schema.indexes
         };
 
-        Object.keys(fields).forEach((fieldName) => {
+        Object.keys(fields).forEach(fieldName => {
             const field = fields[fieldName];
 
             let rules;
 
             if (field.alternatives) {
                 // For schema using alternatives like triggers table.
                 rules = field.alternatives[0].rules;
+            } else if (field.matches && field.matches[0] && field.matches[0].schema) {
```

Comment:
I don't see any documentation for .children in joi versions 16 and above and change logs doesn't describe much. Although if we do a schema.describe().children. It would be undefined. Here are the examples.
Version: 17.2.1
https://runkit.com/embed/e2xaxt0qkkan

Version: 13.2.0
https://runkit.com/glenpike/joi-nested-objects-test-case-1485

pattern.matches example:
https://joi.dev/api/?v=16.1.8#objectpatternpattern-schema-options

Reply:
from what I see from the function doc it's purpose is to expose internal configuration. Wish `joi` had documented it better. As long as it's output is able to provide correct `length` data we are good.